From 7019396c20d775d7d25ae3c89a447d5218a6b6f8 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 10 Nov 2022 13:34:28 +0200
Subject: [PATCH] [SW-105079] common: make pll register offset device specific

Till now the PLL register offset were same across all the asics
(goya, greco, gaudi, gaudi2). These offsets are changed for gaudi3,
so add the PLL offsets in device specific files.

Change-Id: I1ec2461f356daec2804231eafa8e2dc926952093
---

diff --git a/arch/gaudi/arm/hl.h b/arch/gaudi/arm/hl.h
index 4ba5dc0..83365aa 100644
--- a/arch/gaudi/arm/hl.h
+++ b/arch/gaudi/arm/hl.h
@@ -8,6 +8,11 @@
 
 #define BOOT_DESC_MAX_IMGS	5
 
+#define HL_PLL_DIV_FACTOR	0x200
+#define HL_PLL_DIV_FACTOR_CMD	0x220
+#define HL_PLL_DIV_SEL		0x280
+#define HL_PLL_DIV_EN		0x2a0
+
 enum hl_rst_src {
 	HL_COLD_RST,
 	HL_MANUAL_RST,
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
index e24f77c..d3c74a6 100644
--- a/arch/gaudi2/common/hl.h
+++ b/arch/gaudi2/common/hl.h
@@ -11,6 +11,11 @@
 
 #define BOOT_DESC_MAX_IMGS	5
 
+#define HL_PLL_DIV_FACTOR       0x200
+#define HL_PLL_DIV_FACTOR_CMD   0x220
+#define HL_PLL_DIV_SEL	        0x280
+#define HL_PLL_DIV_EN	        0x2a0
+
 void set_pci_msix_pba_offset_val(void);
 void set_slv_cfg_reg_val(void);
 void halt_n_reset_arc(u8 id);
diff --git a/arch/gaudi3/arc/hl.h b/arch/gaudi3/arc/hl.h
index ce0ea44..788fbf0 100644
--- a/arch/gaudi3/arc/hl.h
+++ b/arch/gaudi3/arc/hl.h
@@ -9,6 +9,11 @@
 #define BOOT_DESC_MAX_IMGS	7
 #define READ_BOOT_DESC_FROM_BOOTINFO
 
+#define HL_PLL_DIV_FACTOR	0x30
+#define HL_PLL_DIV_FACTOR_CMD	0x40
+#define HL_PLL_DIV_SEL		0x50
+#define HL_PLL_DIV_EN		0x60
+
 enum gaudi3_rst_src {
 	HL_COLD_RST = 1,
 	HL_MANUAL_RST = 2,
diff --git a/arch/goya/arm/hl.h b/arch/goya/arm/hl.h
index 9bea154..2456cf4 100644
--- a/arch/goya/arm/hl.h
+++ b/arch/goya/arm/hl.h
@@ -8,6 +8,11 @@
 
 #define BOOT_DESC_MAX_IMGS	5
 
+#define HL_PLL_DIV_FACTOR	0x200
+#define HL_PLL_DIV_FACTOR_CMD	0x220
+#define HL_PLL_DIV_SEL		0x280
+#define HL_PLL_DIV_EN		0x2a0
+
 enum hl_rst_src {
 	HL_COLD_RST,
 	HL_MANUAL_RST,
diff --git a/arch/greco/common/hl.h b/arch/greco/common/hl.h
index e498562..6095ec0 100644
--- a/arch/greco/common/hl.h
+++ b/arch/greco/common/hl.h
@@ -44,6 +44,11 @@
 #define BOOT_ROM_SIZE			SZ_64K
 #define FW_SRAM_SIZE			SZ_64K
 
+#define HL_PLL_DIV_FACTOR		0x200
+#define HL_PLL_DIV_FACTOR_CMD		0x220
+#define HL_PLL_DIV_SEL			0x280
+#define HL_PLL_DIV_EN			0x2a0
+
 enum hl_rst_src {
 	HL_COLD_RST,
 	HL_MANUAL_RST,
diff --git a/lib/common/pll.c b/lib/common/pll.c
index 3f3b778..30159e7 100644
--- a/lib/common/pll.c
+++ b/lib/common/pll.c
@@ -8,6 +8,7 @@
 #include "preboot.h"
 #include "uart.h"
 #include "pll.h"
+#include "hl.h"
 #include <device.h>
 #include <bitops.h>
 
diff --git a/lib/common/pll.h b/lib/common/pll.h
index 3c67dd2..101c384 100644
--- a/lib/common/pll.h
+++ b/lib/common/pll.h
@@ -15,10 +15,6 @@
 #define HL_PLL_NR		0x100
 #define HL_PLL_NF		0x104
 #define HL_PLL_OD		0x108
-#define HL_PLL_DIV_FACTOR	0x200
-#define HL_PLL_DIV_FACTOR_CMD	0x220
-#define HL_PLL_DIV_SEL		0x280
-#define HL_PLL_DIV_EN		0x2a0
 #define SYS_REF_CLOCK		50000000ULL
 #define USEC_PER_SEC		1000000
 #define BYPASS_CYCLES_PER_USEC	(SYS_REF_CLOCK / USEC_PER_SEC)
